board/solidrun/mx6cuboxi/mx6cuboxi.c

```diff
@@ -875,4 +877,113 @@ void board_init_f(ulong dummy)
 	/* load/boot image from boot device */
 	board_init_r(NULL, 0);
 }
+
+#ifdef CONFIG_SPL_BOARD_PROVISION
+
+struct srkfdt {
+	uint32_t srk0;
+	uint32_t srk1;
+	uint32_t srk2;
+	uint32_t srk3;
+	uint32_t srk4;
+	uint32_t srk5;
+	uint32_t srk6;
+	uint32_t srk7;
+};
+
+void spl_board_provision() {
+	u32 val, i;
+	u32 mac0, mac1;
+	char c;
+	char input[100];
+
+	hab_rvt_report_status_t *hab_rvt_report_status;
+	hab_rvt_report_status = (hab_rvt_report_status_t *)HAB_RVT_REPORT_STATUS;
+	enum hab_status status = 0;
+	enum hab_config config = 0;
+	enum hab_state state = 0;
+	status = hab_rvt_report_status(&config, &state);
+
+	printf("\nHAB Configuration: 0x%02x, HAB State: 0x%02x, HAB Status: 0x%02x\n",
+		config, state, status);
+
+	// System has secure booted, no need to run the provisioning flow
+	if (state == HAB_STATE_TRUSTED || state == HAB_STATE_SECURE) {
+		return;
+	}
+
+	const void* myfdt = (void*) gd_fdt_blob();
+	int srkhnode = fdt_subnode_offset(myfdt, 0, "srkh");
+	struct srkfdt *srkh = fdt_getprop(myfdt, srkhnode, "srkh-fuse", NULL);
```

Comment:
Should double-check that declaring a multiple u32 array property is okay from a device tree specification standpoint.

Reply:
Looks like it's fine, there's a generic <prop-encoded-array> definition. The Reg property for example uses a bus specific number of u32 pairs in an array.